readme.txt
=== fewcrew ===

fewcrew WordPress Theme, Copyright 2018 Digitoimisto Dude Oy, Morgan Segura
fewcrew is distributed under the terms of the MIT License

Bundled header image, Copyright Brady Bellini
License: CC0 1.0 Universal (CC0 1.0) Public Domain Dedication
Source: https://stocksnap.io/photo/nature-mountains-3T1GEXSD0R

Contributors: Digitoimisto Dude Oy
Tags: one-column, accessibility-ready, translation-ready

Requires at least: 5.9
Tested up to: 6.0.0
Stable tag: 9.0.2
License: MIT License
License URI: https://opensource.org/licenses/MIT

A starter theme called fewcrew.

== Description ==

Air starter theme is built to be very straightforward, front end developer friendly and only partly modular by its structure.

== Installation ==

1. In your admin panel, go to Appearance > Themes and click the Add New button.
2. Click Upload Theme and Choose File, then select the theme's .zip file. Click Install Now.
3. Click Activate to use your new theme right away.

Or just get from GitHub: https://github.com/digitoimistodude/fewcrew

== Frequently Asked Questions ==

= Does this theme support any plugins? =

Yes. There's also a extender, helper plugin Air Helper: https://github.com/digitoimistodude/air-helper

== Changelog ==

= 1.0 - 29 Jan 2016 =
* Initial release

== Credits ==

* Based on Underscores https://underscores.me/, (C) 2012-2017 Automattic, Inc., [GPLv2 or later](https://www.gnu.org/licenses/gpl-2.0.html)
* normalize.css https://necolas.github.io/normalize.css/, (C) 2012-2016 Nicolas Gallagher and Jonathan Neal, [MIT](https://opensource.org/licenses/MIT)


CHANGELOG.md
### 9.0.2: 2022-05-27

- Fix critical regression about ACF blocks not outputting, Fixes #153

### 9.0.1: 2022-05-26

- Disable dropdown box-shadow by default
- Improve placeholder for new project
- Fix typo in CSS variable name
- Fix critical JS error with calling getChildAltText() in vain
- Fix outdated release command

### 9.0.0: 2022-05-24

#### Major update

- Rebranding: Update default logo
- Deprecate %screen-reader-text, use mixin instead
- Deprecate old lazy load styles from future themes
- Add var(--box-shadow-sub-menu) for desktop navigation
- Add more clear starting point with demo content placeholder
- Add support for anchors in the block (thanks @timiwahalahti)
- Disable support for additional CSS classes in blocks (thanks @timiwahalahti)
- Bump tested up WordPress version to 6.0.0
- Fix back to top label for screen readers
- Add data-version attribute to footer indicating fewcrew version in use

### 8.4.7: 2022-05-24

- Remove \_slick.scss, deprecate slick-carousel slider (carousel block is part of [air-blocks](https://github.com/digitoimistodude/air-blocks) now)

### 8.4.6: 2022-05-23

- Combine media queries
- Deprecate heading-hero component

### 8.4.5: 2022-05-20

- Use only one burger animation, revert burger style on mobile

### 8.4.4: 2022-05-20

- Simplify external link selector
- Add js-trigger to skip-link
- Combine nav-mobile media queries
- Update stylelintrc rules block-closing-brace-newline-after and at-rule-empty-line-before
- Fix cubic-bezier mixin $duration variable
- Fix iOS 15 nav width style bug
- Fix iOS 15 scrolling bug with `height: -webkit-fill-available;`
- New navigation animation: slide (default), change to fade by setting `$nav-slide-animation: false;`

### 8.4.3: 2022-05-13

- Fixes #134, nav-toggle menu position shifting on iOS 15.1

### 8.4.2: 2022-05-12

- Add anchors as JS module
- Deprecate vanilla-lazyload and support for older JS-based lazy loading methods
- Drop jQuery
- Rewrite accessible back to top functionality and styles
- Combine external link labels with external link JS module
- Refactor front-end.js

### 8.4.1: 2022-05-12

- Fix paddings on editor
- Remove deprecated text columns block
- Remove default list-type from typography scss and use core-list instead
- Fix responsive lists on gutenberg editor

### 8.4.0: 2022-05-12

- Fix cover block default text color
- Fix overflow in stacked blocks
- Fix core-list not working properly on Gutenberg editor preview
- Fix navigation dropdown menus opening inside viewport after resizing window
- Fix code block style bugs
- Fix pullquote blockquote alignment
- Fix naming of core-button to core-buttons
- Use var(--color-blockquote) color variable in pull quotes
- Simplify core-list block
- Smaller external link icon
- Remove deprecated core-gallery styles, use Gutenberg core styles instead
- Scroll to top when triggering mobile navigation to ensure no gaps are between header and navigation (thanks @Tumppex!)
- Simplify link formatting

### 8.3.8: 2022-04-29

- Fix variable name in link formatting (--color-link-text-hover)
- Add proxyUrl (devpackages 2.4.5)
- Fix config.styles.src that causes gutenberg styles not to compile (devpackages 2.4.6)

### 8.3.7: 2022-04-11

- Remove leftover code from Gutenberg editor
- Move externalLinkDomains list to theme settings for consistency (devpackages 2.4.4)
- Add cssnano and related postcss-plugins, deprecate gulp-clean-css that is in maintenance-mode (devpackages 2.4.4)
- Improve watch task to be more performant (devpackages 2.4.4)
- Add gulp-size and verbose information to console (devpackages 2.4.4)
- Add instructions on how to contribute (devpackages 2.4.4)
- Update stylelint-config-standard, stylelint-config-recommended-css, caniuse-lite and js related packages (devpackages 2.4.4)
- Get phpcs task from devpackages 2.4.4
- Bump tested up WordPress version to 5.9.3

### 8.3.6: 2022-03-31

- Update browser-sync to 2.27.9 (devpackages 2.4.3)
- Fix main title width in article editor
- Add fonts and font-smoothing in article view of the Gutenberg editor

### 8.3.5: 2022-03-25

- Fix: Margin reset breaks article blocks in editor, limit it for ACF blocks only

### 8.3.4: 2022-03-16

- Make edit-post-visual-editor\_\_post-title-wrapper background to match WordPress brand color
- Bump tested up WordPress version to 5.9.2

### 8.3.3: 2022-03-08

- Upgrade to [devpackages 2.4.2](https://github.com/digitoimistodude/devpackages/releases/tag/2.4.2) (details below)
- Upgrade stylelint to 14.5.3
- Update .stylelintrc rules as per the [official recommendations](https://github.com/stylelint/stylelint-config-recommended/issues/157#issuecomment-1056967465)
- Use kebab-kase in fontface mixin from now on
- Bump tested up WordPress version to 5.9.1
- Upgrade eslint to 8.10.0
- Upgrade eslint-config-airbnb to 19.0.4
- Remove deprecated babel-eslint and use @babel/eslint-parser instead
- Revove outdated and unmaintained gulp-eslint and use gulp-eslint-new instead
- Allow js/src/front-end.js to be linted, fix file for JS warnings

### 8.3.2: 2022-02-10

- Reset blockquote margins as it breaks the auto layout in some situations
- Deprecate hiddentext() mixin that is not used anymore

### 8.3.1: 2022-02-08

#### Major update

- Support new native_lazyload_tag() from air-helper
- Add edit link to page
- Move edit link to its own template tag function
- Clean up page.php
- Always allow loading ACF block on preview
- Add a filter to allow modidying ACF block cache keys
- Deprecate default hero template part as it's never used (it's recommended to use a Gutenberg block for it)
- Remove archive.php (it's rarely used, it's similar to index.php and when used, it's always customized anyway)
- Change var(--form-gap) to support both grid-row-gap and grid-column-gap

### 8.3.0: 2022-02-02

- This updates removes some fonts that should not be there, sorry about that

### 8.2.9: 2022-02-02

- Add more weights for Inter font family, fix issue with font file capitalizations and rendering


functions.php
<?php
/**
 * Gather all bits and pieces together.
 * If you end up having multiple post types, taxonomies,
 * hooks and functions - please split those to their
 * own files under /inc and just require here.
 *
 * @Date: 2019-10-15 12:30:02
 * @Last Modified by:   Morgan Segura
 * @Last Modified time: 2022-05-27 10:37:08
 *
 * @package fewcrew
 */

namespace Few_Crew;

/**
 * The current version of the theme.
 */
define( 'Few_Crew_VERSION', '1.0.0' );

/**
 * Theme settings
 */
// We need to have some defaults as comments or empties so let's allow this:
// phpcs:disable Squiz.Commenting.InlineComment.SpacingBefore, WordPress.Arrays.ArrayDeclarationSpacing.SpaceInEmptyArray
add_action( 'after_setup_theme', function() {
  $theme_settings = [
    /**
     * Theme textdomain
     */
    'textdomain' => 'fewcrew',

    /**
     * Image and content sizes
     */
    'image_sizes' => [
      'small'   => 414,
      'medium'  => 768,
      'large'   => 1024,
      'xlarge'   => 1800,
    ],
    'content_width' => 1800,

    /**
     * Logo and featured image
     */
    'default_featured_image'  => null,
    'logo'                    => '/svg/fewcrew-logo.svg',
    'width'                   => null,

    /**
     * Custom setting group settings when using Air setting groups plugin.
     * On multilingual sites using Polylang, translations are handled automatically.
     */
    'custom_settings' => [
      // 'your-custom-setting' => [
      //   'id' => Your custom setting post id,
      //   'title' => 'Your custom setting',
      //   'block-editor' => true,
      //  ],
    ],

    'social_media_accounts'  => [
      // 'twitter' => [
      //   'title' => 'Twitter',
      //   'url'   => 'https://twitter.com/digitoimistodude',
      // ],
    ],

    /**
     * All links are cheked with JS, if those direct to external site and if,
     * indicator of that is included. Exclude domains from that check in this array.
     */
    'external_link_domains_exclude' => [
      // 'localhost:3000',
      // 'airdev.test',
      // 'airwptheme.com',
    ],

    /**
     * Menu locations
     */
    'menu_locations' => [
      'header_primary' => __( 'Account Menu', 'fewcrew' ),
      'header_secondary' => __( 'Sales Menu', 'fewcrew' ),
      'header_tertiary' => __( 'Navigation Menu', 'fewcrew' ),
      'footer' => __( 'Footer Menu', 'fewcrew' ),
    ],

    /**
     * Taxonomies
     *
     * See the instructions:
     * https://github.com/digitoimistodude/fewcrew#custom-taxonomies
     */
    'taxonomies' => [
      // 'your-taxonomy' => [
      //   'name' => 'Your_Taxonomy',
      //   'post_types' => [ 'post', 'page' ],
      // ],
    ],

    /**
     * Post types
     *
     * See the instructions:
     * https://github.com/digitoimistodude/fewcrew#custom-post-types
     */
    'post_types' => [
      // 'your-post-type' => 'Your_Post_Type',
    ],

    /**
     * Gutenberg -related settings
     */
    // Register custom ACF Blocks
    // 'acf_blocks' => [
      // [
        // 'name'           => 'swiper',
        // 'title'          => 'Hero Swiper',
        // You can safely remove lines below if you find no use for them
        // 'prevent_cache'  => false, // Defaults to false,
        // Icon defaults to svg file inside svg/block-icons named after the block name,
        // eg. svg/block-icons/block-file-slug.svg
        //
        // Icon setting defines the dashicon equivalent: https://developer.wordpress.org/resource/dashicons/#block-default
        // 'icon'  => 'block-default',
      // ],
    // ],

    // Custom ACF block default settings
    'acf_block_defaults' => [
      'category'          => 'fewcrew',
      'mode'              => 'auto',
      'align'             => 'full',
      'post_types'        => [
        'page',
      ],
      'supports'  => [
        'align'           => false,
        'anchor'          => true,
        'customClassName' => false,
      ],
      'render_callback'   => __NAMESPACE__ . '\render_acf_block',
    ],

    // Restrict to only selected blocks
    // Set the value to 'all' to allow all blocks everywhere
   'allowed_blocks' => [
      'default' => [
      ],
      'post' => [
        'core/archives',
        'core/audio',
        'core/buttons',
        'core/categories',
        'core/code',
        'core/column',
        'core/columns',
        'core/coverImage',
        'core/embed',
        'core/file',
        'core/freeform',
        'core/gallery',
        'core/heading',
        'core/html',
        'core/image',
        'core/latestComments',
        'core/latestPosts',
        'core/list',
        'core/more',
        'core/nextpage',
        'core/paragraph',
        'core/preformatted',
        'core/pullquote',
        'core/quote',
        'core/block',
        'core/separator',
        'core/shortcode',
        'core/spacer',
        'core/subhead',
        'core/table',
        'core/textColumns',
        'core/verse',
        'core/video',
      ],
    ],

    // If you want to use classic editor somewhere, define it here
    'use_classic_editor' => [],

    // Add your own settings and use them wherever you need, for example THEME_SETTINGS['my_custom_setting']
    'my_custom_setting' => true,
  ];

  $theme_settings = apply_filters( 'Few_Crew_theme_settings', $theme_settings );

  define( 'THEME_SETTINGS', $theme_settings );
} ); // end action after_setup_theme


/**
 * Theme Custom Image Sizes
 */
add_theme_support('post-thumbnails');
add_image_size('post-thumbnail size', 800, 240);
add_image_size('homepage-thumb size', 220, 180);
add_image_size('fullpage-thumb size', 590, 790);


function funny($arr=[], $options='') {
  if ($options == 'dump') {
    return '<pre>' . var_dump($arr) . '</pre>';
  } else {
    return '<pre>' . print_r($arr) . '</pre>';
  }
}

/**
 * Required files
 */
require get_theme_file_path( '/inc/hooks.php' );
require get_theme_file_path( '/inc/includes.php' );
require get_theme_file_path( '/inc/template-tags.php' );

// Run theme setup
add_action( 'init', __NAMESPACE__ . '\theme_setup' );
add_action( 'after_setup_theme', __NAMESPACE__ . '\build_theme_support' );
